torchmetrics/detection/map.py

```diff
@@ -12,87 +12,74 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import logging
-import sys
-from dataclasses import dataclass
-from typing import Any, Callable, Dict, List, Optional, Sequence, Union
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple
 
 import torch
 from torch import Tensor
 
 from torchmetrics.metric import Metric
-from torchmetrics.utilities.imports import (
-    _PYCOCOTOOLS_AVAILABLE,
-    _TORCHVISION_AVAILABLE,
-    _TORCHVISION_GREATER_EQUAL_0_8,
-)
+from torchmetrics.utilities.imports import _TORCHVISION_AVAILABLE, _TORCHVISION_GREATER_EQUAL_0_8
 
 if _TORCHVISION_AVAILABLE and _TORCHVISION_GREATER_EQUAL_0_8:
-    from torchvision.ops import box_convert
+    from torchvision.ops import box_area, box_convert, box_iou
 else:
     box_convert = None
-
-if _PYCOCOTOOLS_AVAILABLE:
-    from pycocotools.coco import COCO
-    from pycocotools.cocoeval import COCOeval
-else:
-    COCO, COCOeval = None, None
+    box_iou = None
+    box_area = None
 
 log = logging.getLogger(__name__)
 
 
-@dataclass
-class MAPMetricResults:
-    """Dataclass to wrap the final mAP results."""
+class BaseMetricResults(dict):
+    """Base metric class, that allows fields for pre-defined metrics."""
 
-    map: Tensor
-    map_50: Tensor
-    map_75: Tensor
-    map_small: Tensor
-    map_medium: Tensor
-    map_large: Tensor
-    mar_1: Tensor
-    mar_10: Tensor
-    mar_100: Tensor
-    mar_small: Tensor
-    mar_medium: Tensor
-    mar_large: Tensor
-    map_per_class: Tensor
-    mar_100_per_class: Tensor
+    def __getattr__(self, key: str) -> Tensor:
+        if key in self:
+            return self[key]
+        else:
+            raise AttributeError(f"No such attribute: {key}")
```

Comment:
isn't this default behavior?

Reply:
Using this you get the correct error message, an `AttributeError` instead of a `KeyError`

Reply:
I see so maybe add a comment to the code, just the same as you told me here...